workflow/controller/operator_test.go

```diff
@@ -11194,6 +11194,8 @@ spec:
               - '-c'
               - sleep 9000
             resources: {}
+            dependencies:
+              - main
```

Comment:
Perhaps add a separate test so that we can ensure the existing tests pass

Reply:
Added. In addition, some optimizations have already been made to the existing test, such as https://github.com/argoproj/argo-workflows/pull/12756#discussion_r1530245007